src/main/java/com/google/roomies/data/Listing.java

```diff
@@ -111,6 +126,8 @@ public static Builder builder() {
     abstract Builder setSinglePrice(Money singlePrice);
     abstract Builder setListingPrice(Money listingPrice);
     abstract Builder setComments(ImmutableList<Comment> comments);
+    abstract Builder setLocation(GeoPoint location);
+    abstract Builder setMilesToCampus(Double milesToCampus);
```

Comment:
You can go with this model just to get the functionality in before the end of the internship, but this ends up being rather hacky and not a model you would want to stay with going forwards. There are a couple of issues here:
- location and miles to campus are strictly redundant information, since you're always using the same location for campus. Which means you also need to be worried about the two getting out of sync.
- If you adopt this model and have some actual listings, then decide you want to support a second campus/location, you have to go fix all of these old listings in storage first

I'm assuming the only reason you are putting the distance information in storage is so that you can do queries by index. For small amounts of data you /could/ probably get away with just loading everything and doing the filtering in memory (this obviously won't scale well). Ultimately you probably want something like the s2 index we talked about.